source/cbor.c

```diff
@@ -0,0 +1,774 @@
+    if (aws_cbor_decoder_peek_type(decoder, &out_type)) {
+        return PyErr_AwsLastError();
+    }
+    if (out_type != AWS_CBOR_TYPE_INDEF_BYTES_START) {
+        aws_raise_error(AWS_ERROR_CBOR_UNEXPECTED_TYPE);
+        return PyErr_AwsLastError();
+    }
+    /* consume the bytes start element */
+    aws_cbor_decoder_consume_next_single_element(decoder);
+    if (aws_cbor_decoder_peek_type(decoder, &out_type)) {
+        return PyErr_AwsLastError();
+    }
+    /* Empty bytes */
+    PyObject *result = PyBytes_FromStringAndSize(NULL, 0);
+    while (out_type != AWS_CBOR_TYPE_BREAK) {
+        PyObject *next_part = s_cbor_decoder_pop_next_bytes_val_to_pyobject(decoder);
+        if (!next_part) {
+            Py_DECREF(result);
+            return NULL;
+        }
+        /* The reference to the old value of bytes will be stolen and next_part will be del. */
+        PyBytes_ConcatAndDel(&result, next_part);
+        if (!result) {
+            return NULL;
+        }
+        if (aws_cbor_decoder_peek_type(decoder, &out_type)) {
+            return PyErr_AwsLastError();
+        }
+    }
+    return result;
+}
+
+/**
+ * helper to get the next inf string
+ */
+static PyObject *s_cbor_decoder_pop_next_inf_string_to_py_str(struct aws_cbor_decoder *decoder) {
+    enum aws_cbor_type out_type = AWS_CBOR_TYPE_UNKNOWN;
+    if (aws_cbor_decoder_peek_type(decoder, &out_type)) {
+        return PyErr_AwsLastError();
+    }
+    if (out_type != AWS_CBOR_TYPE_INDEF_TEXT_START) {
+        aws_raise_error(AWS_ERROR_CBOR_UNEXPECTED_TYPE);
+        return PyErr_AwsLastError();
+    }
+    /* consume the bytes start element */
+    aws_cbor_decoder_consume_next_single_element(decoder);
+    if (aws_cbor_decoder_peek_type(decoder, &out_type)) {
+        return PyErr_AwsLastError();
+    }
+    /* Empty string */
+    PyObject *result = PyUnicode_FromStringAndSize(NULL, 0);
+    while (out_type != AWS_CBOR_TYPE_BREAK) {
+        PyObject *next_part = s_cbor_decoder_pop_next_text_val_to_pyobject(decoder);
+        if (!next_part) {
+            Py_DECREF(result);
+            return NULL;
+        }
+        /* Returns a new reference and keep the arguments unchanged. */
+        PyObject *concat_val = PyUnicode_Concat(result, next_part);
+        Py_DECREF(result);
+        Py_DECREF(next_part);
+        if (!concat_val) {
+            return NULL;
+        }
+        result = concat_val;
+        if (aws_cbor_decoder_peek_type(decoder, &out_type)) {
+            return PyErr_AwsLastError();
+        }
+    }
+    return result;
+}
+
+/**
+ * Generic helper to convert a cbor encoded data to PyObject
+ */
+static PyObject *s_cbor_decoder_pop_next_tag_to_pyobject(struct aws_cbor_decoder *decoder) {
+    uint64_t out_tag_val = 0;
+    if (aws_cbor_decoder_pop_next_tag_val(decoder, &out_tag_val)) {
+        return PyErr_AwsLastError();
+    }
+    /* TODO: implement those tags */
+    switch (out_tag_val) {
```

Comment:
so we currently support no tags? should we be more explcit in doc on what function supports?